File: app/src/main/java/io/github/ketzalv/validationedittext/sample/RegisterFragment.java
package io.github.ketzalv.validationedittext.sample;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

import io.github.ketzalv.validationedittext.ValidationEditText;
import io.github.ketzalv.validationedittext.sample.base.BaseFragment;
import io.github.ketzalv.validationedittext.sample.base.IValidateFormsCycle;
import io.github.ketzalv.validationedittext.sample.utils.AlertUtils;

public class RegisterFragment extends BaseFragment implements IValidateFormsCycle, View.OnClickListener {

    public static final String TAG = "RegisterFragment";

    private List<ValidationEditText> requiredFields = new ArrayList<>();
    private HashMap<String, String> data = new HashMap<>();

    public static RegisterFragment newInstance() {

        Bundle args = new Bundle();

        RegisterFragment fragment = new RegisterFragment();
        fragment.setArguments(args);
        return fragment;
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        if(getView() == null){
            return inflater.inflate(R.layout.fragment_register, container, false);
        }
        return getView();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        initViews(view);
        setupRequiredFields(view);
    }

    private void setupRequiredFields(View view) {
        if(view == null){
            return;
        }
        if(!requiredFields.isEmpty()){
            requiredFields.clear();
        }
        requiredFields.add((ValidationEditText) view.findViewById(R.id.edit_username));
        requiredFields.add((ValidationEditText) view.findViewById(R.id.edit_email));
        requiredFields.add((ValidationEditText) view.findViewById(R.id.edit_name));
        requiredFields.add((ValidationEditText) view.findViewById(R.id.edit_lastname));
        requiredFields.add((ValidationEditText) view.findViewById(R.id.edit_surname));
        requiredFields.add((ValidationEditText) view.findViewById(R.id.edit_birthdate));

    }

    private void initViews(View view) {
        view.findViewById(R.id.button_register).setOnClickListener(this);
    }

    @Override
    public String getFragmentTag() {
        return TAG;
    }

    @Override
    public void setValuesDefaultForm() {

    }

    @Override
    public void onValidateForm() {
        boolean isInvalidForm = false;
        for(ValidationEditText editText: requiredFields){
            if(!editText.isValidField()){
                isInvalidForm = true;
            }
        }
        if(isInvalidForm){
            return;
        }
        onValidationSuccess();
    }

    @Override
    public void onGetData() {
        if(!data.isEmpty()){
            data.clear();
        }
        for(ValidationEditText editText: requiredFields){
            data.put(editText.getHint().toString(), editText.getText().toString());
        }
    }

    @Override
    public void onValidationSuccess() {
        onGetData();
        perfomSendData();
    }

    @Override
    public void onClick(View v) {
        onValidateForm();
    }

    private void perfomSendData() {
        AlertUtils.showGenericAlert(getActivity(),
                true,
                R.string.getdata,
                LoginFragment.convertHashMapToString(data));
    }
}

File: sample/src/main/java/io/github/ketzalv/validationedittext/sample/RegisterFragment.kt
package io.github.ketzalv.validationedittext.sample

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import io.github.ketzalv.validationedittext.ValidationEditText
import io.github.ketzalv.validationedittext.sample.LoginFragment.Companion.convertHashMapToString
import io.github.ketzalv.validationedittext.sample.base.BaseFragment
import io.github.ketzalv.validationedittext.sample.base.IValidateFormsCycle
import io.github.ketzalv.validationedittext.sample.utils.AlertUtils.showGenericAlert
import kotlinx.android.synthetic.main.fragment_register.*
import java.util.*

class RegisterFragment : BaseFragment(), IValidateFormsCycle, View.OnClickListener {
    private val requiredFields: MutableList<ValidationEditText> = ArrayList()
    private val data = HashMap<String, String>()
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return if (view == null) {
            inflater.inflate(R.layout.fragment_register, container, false)
        } else view
    }

    override val fragmentTag: String?
        get() = TAG

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        initViews(view)
        setupRequiredFields(view)
    }

    private fun setupRequiredFields(view: View?) {
        if (view == null) {
            return
        }
        if (!requiredFields.isEmpty()) {
            requiredFields.clear()
        }
        requiredFields.add(edit_username)
        requiredFields.add(edit_email)
        requiredFields.add(edit_name)
        requiredFields.add(edit_lastname)
        requiredFields.add(edit_surname)
        requiredFields.add(edit_birthdate)
    }

    private fun initViews(view: View) {
        button_register?.setOnClickListener(this)
    }

    override fun setValuesDefaultForm() {}
    override fun onValidateForm() {
        var isInvalidForm = false
        for (editText in requiredFields) {
            if (!editText.isValidField) {
                isInvalidForm = true
            }
        }
        if (isInvalidForm) {
            return
        }
        onValidationSuccess()
    }

    override fun onGetData() {
        if (!data.isEmpty()) {
            data.clear()
        }
        for (editText in requiredFields) {
            data[editText.hint.toString()] = editText.text.toString()
        }
    }

    override fun onValidationSuccess() {
        onGetData()
        perfomSendData()
    }

    override fun onClick(v: View) {
        onValidateForm()
    }

    private fun perfomSendData() {
        showGenericAlert(activity,
                true,
                R.string.getdata,
                convertHashMapToString(data))
    }

    companion object {
        const val TAG = "RegisterFragment"
        fun newInstance(): RegisterFragment {
            val args = Bundle()
            val fragment = RegisterFragment()
            fragment.arguments = args
            return fragment
        }
    }
}